src/jsifier.js

```diff
@@ -422,6 +422,19 @@ function JSify(data, functionsOnly) {
     // "Final shape that will be created").
     print('// EMSCRIPTEN_END_FUNCS\n');
 
+    if (ASSERTIONS) print('var ASSERTIONS = true;\n');
+
+    print(preprocess(read('arrayUtils.js')));
+    // Export all arrayUtils.js functions
+    print(maybeExport('intArrayFromString'));
+    print(maybeExport('intArrayToString'));
+
+    if (SUPPORT_BASE64_EMBEDDING) {
+      print(preprocess(read('base64Utils.js')));
+      // Export all base64Utils.js functions
+      print(maybeExport('intArrayFromBase64'));
+      print(maybeExport('tryParseAsDataURI'));
+    }
```

Comment:
Oh, I see that this removes the string functions in the file `src/preamble.js`, and then under the hood `src/jsifier.js` concatenates them back in here.

In general I find a trend where the logic for crafting the output contents of the compilation moves inside the toolchain to be undesirable in comparison to using #ifs and #includes in the input files. I think we should be moving to the opposite direction, where the structure of the compiled output would be readable from the input data files as clearly as possible, instead of being synthesized in jsifier.js or similar.

If the above code could be replaced with the following style of block in `src/preamble.js`, that would be preferable:

```javascript
#if ASSERTIONS
var ASSERTIONS = true;
#endif

#include "arrayUtils.js"
maybeExport('intArrayFromString');
maybeExport('intArrayToString');

#if SUPPORT_BASE64_EMBEDDING
#include "base64Utils.js"
maybeExport('intArrayFromBase64');
maybeExport('tryParseAsDataURI');
#endif
```

this way we would be preserving the final structure of the compiled output more clearly in the JavaScript files, which is easier to read for developers who are trying to understand what constitutes the build output.

If there is some fundamental reason why the above kind of structure did not work, then that's alright, we can do something like that as a followup work. (I know the preprocessor machinery that operates on the JS file is not very robust at the moment)